### ISSUE TITLE
Unable to get items from Large list with filter query

### ISSUE THREAD

**Issue by bhoomesh-spe** [pnp/pnpjs#2742]
Hello Everyone,

I have created a SharePoint list that contains a "lookup" column with the name Status.

The list contains more than 5000 items. I need to filter data based on Status from the BigList.

I have tried with [getAll()](https://pnp.github.io/pnpjs/sp/items/#get-all-items) but it is not working as expected.

Below is my code snippet,

```
import { spfi } from "@pnp/sp";
import "@pnp/sp/webs";
import "@pnp/sp/lists";
import "@pnp/sp/items";
import "@pnp/sp/items/get-all";

const sp = spfi(...);

const allItems: any[] = await sp.web.lists.getByTitle("BigList").items.select("*,Status/Title").expand("Status/Title").filter("Status/Title eq 'Open'").getAll();
console.log(allItems.length);

```

Can anyone help me with the same?

@patrick-rodgers  @juliemturner  @AJIXuMuK @bcameron1231 - Hope you have some solution.

Thanks

**Comment by juliemturner** [comment]
We include a template for submitting an issue that includes sections for the details we need to be able to assist you. It includes sections for expected behavior and observed behavior... 

>I have tried with [getAll()](https://pnp.github.io/pnpjs/sp/items/#get-all-items) but it is not working as expected.

Please elaborate on exactly what you're attempting and what you're seeing that isn't what you expect.

**Comment by bhoomesh-spe** [comment]
@juliemturner Thanks for the quick response.

I need to filter SP list data on the lookup column. In my list, Status is the lookup column.

If I tried with the below code. Everything working as expected.
`const allItems: any[] = await sp.web.lists.getByTitle("BigList").items.select("*,Status/Title").expand("Status/Title").getAll();`

If I tried with the below code. It is not working.
`const allItems: any[] = await sp.web.lists.getByTitle("BigList").items.select("*,Status/Title").expand("Status/Title").filter("Status/Title eq 'Open'").getAll();`

Thanks
